### ISSUE TITLE
Deleting notes inside notebooks crashes tomboy-ng

### ISSUE THREAD

**Issue by uwemock** [tomboy-notes/tomboy-ng#312]
tomboy-ng gets stuck upon deleting notes that are inside a notebook.

To reproduce:
Create a notebook, create a note inside the notebook. Open the new note and klick the delete button ("X" in a circle). Confirm "Yes". tomboy-ng hangs. However, the note will be gone when you restart tomboy-ng.

When I delete notes that are not connected to a notebook, tomboy-ng _might_ crash a little later.

**Comment by davidbannon** [comment]
Yep, I can replicate the main issue. Its clearly been there since I made some substantial changes to the way notes are listed in the right hand list box (to speed up that listing).  I personally don't use the notebook model that much and apparently just did not stumble across that bug. Not very good !

So, not that hard to fix but I  found two more, closely related bugs !  One when adding and immediately deleting a Notebook and the other, a newly created Notebook was not appearing in the notebook list immediately. This area of the code is quite complicated so I need to run some extensive tests before pushing out even a beta.

Thanks [uwemock](https://github.com/uwemock), if you had not spotted this, who know how long it would be in there for !

**Now, the second part of your report,** 

>When I delete notes that are not connected to a notebook, tomboy-ng might crash a little later.

That I am afraid I cannot reproduce. I have deleted over a 100 notes, mostly not members of a notebook and I cannot produce any instability.  Can you tell me a bit more about what you do that might be the trigger for for this problem ?
(just groping in the dark here ...)
Operating system ?
Approx number of notes in your system, hundreds, thousands ?
Do you have a lot of notes open at the time ?
Are these notes that you delete, old, existing notes or brand new ones ?
Anything else you can think of ?

Thanks for an important bug report !

Davo

**Comment by uwemock** [comment]
I cannot fully reproduce the crash when deleting notes that are not member of a notbook. Once I can, I will file another bug.

**Comment by davidbannon** [comment]
OK, I have just released a beta that addresses this issue. Reasonably well tested and probably quite safe to use.
https://github.com/tomboy-notes/tomboy-ng/releases/tag/v0.40a

While I have addressed the first part of your report, I am not so sure about the second part. I suspect it might relate to a situation where an empty note was previously deleted and then an existing one opened, I found potential issues in that space but could not be sure. Feel free to let me know your experiences !

Thanks for your report and helping to make tomboy-ng a better product.

Davo
